messaging/src/main/java/org/axonframework/deadline/dbscheduler/DbSchedulerDeadlineManager.java

```diff
@@ -0,0 +1,375 @@
+            InterceptorChain chain = new DefaultInterceptorChain<>(
+                    unitOfWork,
+                    handlerInterceptors(),
+                    interceptedDeadlineMessage -> {
+                        executeScheduledDeadline(interceptedDeadlineMessage,
+                                                 deadlineDetails.getDeserializedScopeDescriptor(serializer));
+                        return null;
+                    });
+            ResultMessage<?> resultMessage = unitOfWork.executeWithResult(chain::proceed);
+            if (resultMessage.isExceptional()) {
+                Throwable e = resultMessage.exceptionResult();
+                span.recordException(e);
+                logger.warn("An error occurred while triggering deadline with name [{}].",
+                            deadlineDetails.getDeadlineName());
+                throw new DeadlineException("Failed to process", e);
+            }
+        }
+    }
+
+    private void executeScheduledDeadline(DeadlineMessage<?> deadlineMessage, ScopeDescriptor deadlineScope) {
+        scopeAwareProvider.provideScopeAwareStream(deadlineScope)
+                          .filter(scopeAwareComponent -> scopeAwareComponent.canResolve(deadlineScope))
+                          .forEach(scopeAwareComponent -> {
+                              try {
+                                  scopeAwareComponent.send(deadlineMessage, deadlineScope);
+                              } catch (Exception e) {
+                                  String exceptionMessage = format(
+                                          "Failed to send a DeadlineMessage for scope [%s]",
+                                          deadlineScope.scopeDescription()
+                                  );
+                                  throw new ExecutionException(exceptionMessage, e);
+                              }
+                          });
+    }
+
+    @Override
+    public void shutdown() {
+        scheduler.stop();
+        deadlineManagerReference.set(null);
+    }
+
+    /**
+     * Builder class to instantiate a {@link DbSchedulerDeadlineManager}.
+     * <p>
+     * The {@link TransactionManager} is defaulted to a {@link NoTransactionManager} and the {@link SpanFactory}
+     * defaults to a {@link NoOpSpanFactory}.
+     * <p>
+     * The {@link JobScheduler}, {@link ScopeAwareProvider} and {@link Serializer} are <b>hard requirements</b> and as
+     * such should be provided.
+     */
+    public static class Builder {
+
+        private Scheduler scheduler;
+        private ScopeAwareProvider scopeAwareProvider;
+        private Serializer serializer;
+        private TransactionManager transactionManager = NoTransactionManager.INSTANCE;
+        private SpanFactory spanFactory = NoOpSpanFactory.INSTANCE;
+
+        /**
+         * Sets the {@link Scheduler} used for scheduling and triggering purposes of the deadlines. It should have this
```

Comment:
The grammar in this javadoc is a bit off

Reply:
Improved and expended the javadoc.